crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -782,88 +778,68 @@ impl Object {
 
                     snapshot_query.union(historical_query)
                 })
-                .optional()
             })
             .await?;
 
-        // For the moment, if the object existed at some point, it will have eventually be written
-        // to objects_snapshot. Therefore, if both results are None, the object has never existed.
-        let Some(stored_objs) = results else {
-            return Ok(None);
-        };
-
         // Select the max by key after the union query, because Diesel currently does not support order_by on union
         stored_objs
             .into_iter()
             .max_by_key(|o| o.object_version)
-            .map(Self::try_from)
+            .map(|obj| {
+                // The object's checkpoint_sequence_number is set as the historical context, so that
+                // its fields are correctly bounded.
```

Comment:
having changed to checkpoint_viewed_at, I think this is also incorrect. It's possible - in fact, will always be the case - that we'll view an object at version at some checkpoint. So this will also need checkpoint_viewed_at information.